src/content-feature.js

```diff
@@ -259,4 +265,126 @@ export default class ContentFeature {
             flag: `${this.name}${suffix}`
         })
     }
+
+    /**
+     * Define a property descriptor. Mainly used for defining new properties. For overriding existing properties, consider using wrapProperty(), wrapMethod() and wrapConstructor().
+     * @param {any} object - object whose property we are wrapping (most commonly a prototype)
+     * @param {string} propertyName
+     * @param {PropertyDescriptor} descriptor
+     */
+    defineProperty (object, propertyName, descriptor) {
+        // make sure to send a debug flag when the property is used
+        // NOTE: properties passing data in `value` would not be caught by this
+        ['value', 'get', 'set'].forEach((k) => {
+            const descriptorProp = descriptor[k]
+            if (typeof descriptorProp === 'function') {
+                // eslint-disable-next-line @typescript-eslint/no-this-alias
+                const self = this
```

Comment:
`self` is shadowing `window.self` kinda shocked the eslint rule isn't telling you off for that too. Could we choose a different name here?

Alternatively you could alias the debug method instead perhaps?

```suggestion
                const addDebugFlag = this.addDebugFlag
```

Reply:
Right, thanks for catching!